objectherkenning_openbare_ruimte/databricks_pipelines/common/table_manager.py

```diff
@@ -0,0 +1,182 @@
+from datetime import datetime
+
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import StructType
+
+
+class TableManager:
+    def __init__(self, spark: SparkSession, catalog: str, schema: str):
+        self.spark = spark
+        self.catalog = catalog
+        self.schema = schema
+
+    def update_status(
+        self, table_name: str, job_process_time: datetime, exclude_ids=[]
+    ):
+        count_pending_query = f"""
+        SELECT COUNT(*) as pending_count
+        FROM {self.catalog}.{self.schema}.{table_name}
+        WHERE status = 'Pending'
+        """  # nosec
+        total_pending_before = self.spark.sql(count_pending_query).collect()[0][
+            "pending_count"
+        ]
+
+        update_query = f"""
```

Comment:
Does this query update everything independently of the run?
Should there not be a filter to update only specific rows?

Reply:
this pipeline runs every 1/2 weeks and all the images from the previous runs have been processed already. 
I implemented it such that when the pipeline finishes all images from the run have been processed. 
if something is still Pending, it means it pipeline failed and it must be re-triggered